patches/net/minecraft/client/Minecraft.java.patch

```diff
@@ -326,14 +326,16 @@
      }
  
      private boolean isLevelRunningNormally() {
-@@ -2056,6 +_,7 @@
+@@ -2056,7 +_,8 @@
      }
  
      public void setLevel(ClientLevel p_91157_, ReceivingLevelScreen.Reason p_341652_) {
+-        this.updateScreenAndTick(new ReceivingLevelScreen(() -> false, p_341652_));
 +        if (this.level != null) net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.level.LevelEvent.Unload(this.level));
-         this.updateScreenAndTick(new ReceivingLevelScreen(() -> false, p_341652_));
++        this.updateScreenAndTick(new ReceivingLevelScreen(() -> false, p_341652_, this.level == null ? null : p_91157_.dimension(), this.level == null ? null : this.level.dimension()));
```

Comment:
Wouldn't it be simpler to register a `BiFunction<BooleanSupplier, ReceivingLevelScreen.Reason, ? extends ReceivingLevelScreen>` (and return `ReceivingLevelScreen::new` instead of null from DimensionTransitionManager hwen there's noting registered)? That class doesn't look like it would present any hurdles to subclassing.

Reply:
@GizmoTheMoonPig, what do you think of this suggestion?

Reply:
Sorry I missed this. I was trying to decipher it last week but was unable to, then I got really busy on a mod port I'm working on. 
I'm afraid I don't quite understand what's being asked here. Might just be my exhausted brain but I can't comprehend what's being asked here at all. I know its stupid but if you could dumb it down a bit I would appreciate it

Reply:
I think the suggestion is to allow modders to instead subclass `ReceivingLevelScreen` and then register constructors for that screen (the `BiFunction` being a functional interface typed with the same arguments as `ReceivingLevelScreen`'s constructor, so `ReceivingLevelScreen::new`), or something along those lines.

Reply:
alright, I think I understand now. I went ahead and pushed a change to the system. Let me know if it looks alright!